### DISCUSSION TITLE
Extend PhpSpreadsheet with a custom function

### DISCUSSION THREAD

**Discussion by robertotremonti** [PHPOffice/PhpSpreadsheet#2118] [Q&A] [answered]
Hi,
is there a way to extend the class in order to add a custom function?

My purpose is to calculate MAX, MIN, AVG, COUNT, ... from a string like "123+456+78+90".

Thanks for your help.

**Comment by MarkBaker** [comment]
Surely this should just be something that you implement in your own code, calling the appropriate functions from PHPSpreadsheet

**Reply by robertotremonti** [reply]
Thank for your reply, @MarkBaker
But currently there's no way to do it in PhpSpreadsheet.
I mean I can't calculate MAX, MIN, AVG, COUNT, SUM and so on from a string using available functions.
My code gets a Excel formula and data as input from user, so it's not a fixed situation.

**Reply by robertotremonti** [reply]
LEFT, FIND and RIGHT are the only methods to get substrings and treat them as int. I can't parse string to get an array and make some operations with the items AFAICS.

**Comment by MarkBaker** [comment, marked as answer]
No, you can't calculate MAX, MIN, AVG, COUNT and SUM from a single string in MS Excel; what you'd have to do is split that string into an array of values, which is much easier done in PHP than in Excel; then pass that array as an argument to the relevant Excel function implementations to get the results.

```php
$inputString = "123+456+78+90";
$dataValues = explode('+', $inputString);

$max = \PhpOffice\PhpSpreadsheet\Calculation\Statistical\Maximum::max(...$dataValues);
$min = \PhpOffice\PhpSpreadsheet\Calculation\Statistical\Minimum::min(...$dataValues);
...etc
```
